### PR TITLE
SSH to centos IPA node documentation

### PR DESCRIPTION
**Pull request by smoshiur1237**
This documentation describes necessary steps to enable SSH to CentOS IPA node using dynamic-login feature from IPA image.

### CODE REVIEW

**Review by Rozzii — CHANGES_REQUESTED** [review]
I have made a few comments.

File: centos-IPA-ssh/centos-IPA-ssh.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+CCD requests to investigate the possibility to SSH to centos IPA node using dynamic-login feature in IPA image builing. Here the SSH key is given to the ironic-api and ironic-conductor which eventually pass it to the IPA node and enable the user to ssh to the IPA node.
+
+
+## Setup Environment and Testing
+
+### Step 1
+The investigation was performed in Metal3 development environment where we had bare metal host in ready state. `Ironic-api` and `ironic-conductor` containers are running as a docker container. After `docker exec`, first install an editor in the container to edit the ironic.conf.j2 file to add the ssh-key in both containers.
+       
+       docker exec -it ironic-api /bin/bash
+       yum update
```
[the remark this answers is not in the document]
Comment (smoshiur1237) [reply centos-IPA-ssh/centos-IPA-ssh.md]: updated done.

File: centos-IPA-ssh/centos-IPA-ssh.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# Enable SSH to centos IPA node using dynamic-login feature
+
+## Investigation status 12.10.2021
+
+The task requires to investigate the possibility to SSH from host to centos IPA node using dynamic-login feature in IPA image builing. Here the SSH key is given to the ironic-api and ironic-conductor which eventually pass it to the IPA node and enable the user to do the ssh to the IPA node.
```
Comment (Rozzii) [inline centos-IPA-ssh/centos-IPA-ssh.md:5]: The task requires to investigate -> this documentation describes the ....

File: centos-IPA-ssh/centos-IPA-ssh.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+## Setup Environment and Testing
+
+### Step 1
+The investigation was performed in Metal3 development environment where we had bare metal host in ready state. `Ironic-api` and `ironic-conductor` containers are running as a docker container. After `docker exec`, first install an editor in the container to edit the ironic.conf.j2 file to add the ssh-key in both containers.
```
Comment (Rozzii) [inline centos-IPA-ssh/centos-IPA-ssh.md:10]: The investigation -> The steps in the examples were ...

Comment (Rozzii) [inline centos-IPA-ssh/centos-IPA-ssh.md:10]: environment where we -> environment that had

File: centos-IPA-ssh/centos-IPA-ssh.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+Edit `ironic.conf.j2` file resides in `/etc/ironic/` in the following line:
+
+       pxe_append_params = nofb nomodeset vga=normal ipa-insecure=1 {% if env.IRONIC_RAMDISK_SSH_KEY %}sshkey="{{ env.IRONIC_RAMDISK_SSH_KEY|trim }}"{% endif %} {{ env.IRONIC_KERNEL_PARAMS|trim }}
+
+Add user's host `sshkey` inside double quote`("")` from `.ssh/id_rsa.pub` file and edit the line:
+
+       pxe_append_params = nofb nomodeset vga=normal ipa-insecure=1 sshkey="ssh-rsa ...." {{ env.IRONIC_KERNEL_PARAMS|trim }}
+
+It is alos possible to pass that sshkey using the envirnment variable. Now restart the `ironic-api` container. Also user should to do the same step with `ironic-conductor` container and restart that container too. After restart verify the `sshkey` from `/etc/ironic/ironic.conf` file in both containers. The ironic.conf file will get new `sshkey` from `ironic.conf.j2` template file.
```
Comment (Rozzii) [inline centos-IPA-ssh/centos-IPA-ssh.md:24]: Now restart -> The next step is to restart

File: centos-IPA-ssh/centos-IPA-ssh.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# Enable SSH to centos IPA node using dynamic-login feature
+
+## Investigation status 12.10.2021
```
Comment (Rozzii) [inline centos-IPA-ssh/centos-IPA-ssh.md:3]: Investigation status 12.10.2021 - > Goal of the documentation
I think there is no need to write date as the document is version controlled.

**Review by Rozzii — CHANGES_REQUESTED** [review]
I have one last comment.

**Review by Rozzii — APPROVED** [review]

**Pull request merged**